presto-kafka/src/main/java/io/prestosql/plugin/kafka/encoder/json/format/CustomDateTimeFormatFunctions.java

```diff
@@ -0,0 +1,91 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.plugin.kafka.encoder.json.format;
+
+import org.joda.time.DateTime;
+import org.joda.time.DateTimeZone;
+import org.joda.time.Instant;
+import org.joda.time.format.DateTimeFormat;
+import org.joda.time.format.DateTimeFormatter;
+
+import java.util.Locale;
+import java.util.function.Function;
+
+import static io.prestosql.spi.type.DateTimeEncoding.unpackMillisUtc;
+import static io.prestosql.spi.type.DateTimeEncoding.unpackZoneKey;
+import static java.lang.String.format;
+
+public final class CustomDateTimeFormatFunctions
+{
+    private CustomDateTimeFormatFunctions() {}
+
+    public static Function<Long, String> formatDateFunc(String formatHint)
+    {
+        try {
+            DateTimeFormatter formatter = DateTimeFormat.forPattern(formatHint).withLocale(Locale.ENGLISH).withZoneUTC();
+            return millis -> (new DateTime(millis, DateTimeZone.UTC)).toString(formatter);
```

Comment:
I think it would be better to supplement the session zone here.
This would make it more similar to the non-legacy case, when we implement it.

Ie 
`INSERT ,.. TIMESTAMP 'some_date_time'`
would be `some_date_time session_zone` in legacy
and `some_date_time` in non-legacy

and if the format hint does not contain the zone,
they would actually be the same.